src/envs/behavior.py

```diff
@@ -329,8 +333,32 @@ def current_ig_state_to_state(self) -> State:
                 ig_obj.get_orientation(),
             ])
             state_data[obj] = obj_state
-        simulator_state = self.igibson_behavior_env.task.save_scene()
-        return State(state_data, simulator_state)
+
+        def save_unique_checkpoint(sim: "Simulator",
+                                   save_dir: str) -> Tuple[str, str]:
+            unique_id = int(np.random.randint(size=1, low=0, high=10**6))
+            path = save_dir + str(unique_id) + "/"
+            isExist = os.path.exists(path)
+            if not isExist:
+                os.makedirs(path)
+            simulator_state = save_checkpoint(sim, path)
+            return str(simulator_state), str(unique_id)
```

Comment:
I just took another look here and noticed the `str(simulator_state)`. what's going on there? isn't `simulate_state` like a really large and gross dictionary?